### PR TITLE
Fix incorrect prediction result for TensorFlow

### LINKED ISSUE #1397: Incorrect evaluation results when training model.
Default metrics evaluation results are incorrect when using tf.estimator.DNNClassifer for binary classification prediction.
I have reproduced this problem on the iris dataset，is shown in the following zip file.
[iris-dnn.html.zip](https://github.com/sql-machine-learning/sqlflow/files/3950119/iris-dnn.html.zip)

### PR DESCRIPTION
**Pull request by tonyyang-svail**
Fix #1397

### CODE REVIEW

File: python/sqlflow_submitter/tensorflow/predict.py [hunk 1]
```diff
@@ -50,35 +50,15 @@ def define_tf_flags():
 class FastPredict:
     def __init__(self, estimator, input_fn):
         self.estimator = estimator
-        self.first_run = True
-        self.closed = False
         self.input_fn = input_fn
 
-    def _create_generator(self):
```
Comment (tonyyang-svail) [inline python/sqlflow_submitter/tensorflow/predict.py:57]: @typhoonzero I don't understand why we need to use `self.next_feature` to access features. Please be aware that TF launches `_create_generator` in a separate thread, while `self.next_feature` introduces a sharing state, which is error-prone.

File: python/sqlflow_submitter/tensorflow/predict.py [hunk 2]
```diff
@@ -50,35 +50,15 @@ def define_tf_flags():
...
-        while not self.closed:
```
Comment (tonyyang-svail) [inline python/sqlflow_submitter/tensorflow/predict.py:58]: @typhoonzero I don't understand why we need the `while loop`. I guess is that `len(feature_batch) == 2` BUG below led the implementer add it.

File: python/sqlflow_submitter/tensorflow/predict.py [hunk 3]
```diff
@@ -50,35 +50,15 @@ def define_tf_flags():
...
-            yield self.next_features[0], self.next_features[1]
-
-    def predict(self, feature_batch):
-        self.next_features = feature_batch
-        if self.first_run:
-            self.batch_size = len(feature_batch)
```
Comment (tonyyang-svail) [inline python/sqlflow_submitter/tensorflow/predict.py:64]: @typhoonzero BUG: the name `feature_batch` is misleading. It actually is a `(feature, label)` tuple with the batch size is always 1. However, `len(feature_batch) == 2`, and we shouldn't assign it to `self.batch_size`.

**Review by typhoonzero — APPROVED** [review]
LGTM, FYI. previous implementation: https://github.com/sql-machine-learning/sqlflow/pull/555, the `FastPredict` is completely copied from: https://github.com/marcsto/rl/blob/master/src/fast_predict2.py, mentioned by: https://guillaumegenthial.github.io/serving-tensorflow-estimator.html#a-clever-fix

And seems Estimator have many limitations to use, if we really need a better prediction python code, I'll recomment we do not use estimator.
